src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -375,6 +375,9 @@ private static (MethodInfo, Type[]) GetCachedMethodInfo(AssemblyKey assemblyKey,
         "ReflectionAnalysis",
         "IL2060:MakeGenericMethod",
         Justification = "https://github.com/mono/linker/issues/1727")]
+    [SuppressMessage("AOT",
+        "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.",
+        Justification = "Methods being referenced do not have DynamicallyAccessedMembers.")]
```

Comment:
1. This isn't a valid justification for suppressing this warning.
2. I don't believe this warning can be safely suppressed. From looking at the only callsite, this is used to convert a `ValueTask<T>` to a `Task<T>`. If the `T` is a ValueType, this code isn't guaranteed to work in Native AOT, since `CreateValueTaskConverter<T>` might not be generated ahead of time for that specific ValueType.

Reply:
Further, how important is it for Microsoft.JSInterop to be AOT-compatible in .NET 8? Isn't it only used for WASM, which doesn't use Native AOT?

Reply:
It is used across WASM, Server and WebView. You've raised important points about these suppressions, and I will either pursue a fallback like you describe [here](https://github.com/dotnet/aspnetcore/pull/45580#discussion_r1048741270) or document my findings in an issue report for us to milestone and discuss like I did with BindConverter (#45578).

Reply:
Documented in #45795